### PR TITLE
add/edit subjects to course

### PR DESCRIPTION
**Pull request by MuhammadTamzid**
Dear Sir,
created a new course with selected subjects, using nested attributes form_for course and fields_for course_subjects.

Thank you (bow)

### CODE REVIEW

File: app/models/course_subject.rb [hunk 1]
```diff
@@ -1,4 +1,16 @@
 class CourseSubject < ActiveRecord::Base
   belongs_to :course
   belongs_to :subject
+
+  validates :course_id, uniqueness: {scope: :subject_id}
+
+  after_save :create_user_subject
+
+  private
+  def create_user_subject
```
Comment (chuanhtuan) [inline app/models/course_subject.rb]: user_subject will be created when supervisor click Start Subject
remove in this pull request and make a new pull request for this feature

File: app/views/supervisor/course_subjects/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<% provide(:title, 'All Subjects') %>
```
Comment (chuanhtuan) [inline app/views/supervisor/course_subjects/show.html.erb]: remove this function
add subject will be nested in new or edit Course

File: app/views/supervisor/courses/_course.html.erb [hunk 3]
```diff
@@ -1,11 +1,15 @@
  <tr>
     <td><%= link_to course.name, [:supervisor, course] %></td>
-    <td>(0) <%= link_to t(:add_edit_text), '#' %></td>
+    <td><%= course.description.truncate 50 %></td>
+    <td>
+        (<%= course.subjects.count %>)
+        <%= link_to t(:add_edit_text), supervisor_course_course_subjects_path(course) %>
```
Comment (chuanhtuan) [inline app/views/supervisor/courses/_course.html.erb]: nested course_subject, it means you have to use nested attribute for this feature

File: app/controllers/supervisor/course_subjects_controller.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+class Supervisor::CourseSubjectsController < ApplicationController
+  before_action :load_course, only: [:update, :show]
+
+  def show
+    @subjects = Subject.all
+  end
+
+  def update
```
Comment (chuanhtuan) [inline app/controllers/supervisor/course_subjects_controller.rb]: remove this function
add subject will be nested in new or edit Course

File: app/models/course_subject.rb (hunk 1 above)
Comment (MuhammadTamzid) [inline app/models/course_subject.rb]: will remove

File: app/views/supervisor/courses/_course.html.erb (hunk 3 above)
Comment (MuhammadTamzid) [inline app/views/supervisor/courses/_course.html.erb]: Sir, can we use this?
"link_to display all subjects" 
<%= link_to t(:add_edit_text), supervisor_course_course_subjects_path(course) %>
then in supervisor-> course_subjects->show.html.erb
<%= form_for @course, url: supervisor_course_path(@course, type: "update_subjects") do |f| %>
.
.
<%= collection_check_boxes :course, :subject_ids, @subjects, :id, :name do |builder| %>
.
.
<% end %>
.
.
<% end %>

File: app/controllers/supervisor/course_subjects_controller.rb (hunk 4 above)
Comment (MuhammadTamzid) [inline app/controllers/supervisor/course_subjects_controller.rb]: will remove and will implement in nested in course

File: app/views/supervisor/courses/_course.html.erb (hunk 3 above)
Comment (chuanhtuan) [inline app/views/supervisor/courses/_course.html.erb]: we should create a new course with selected subjects as like created new subject with tasks

File: app/models/course.rb [hunk 5]
```diff
@@ -6,9 +14,35 @@ class Course < ActiveRecord::Base
   validates :end_date, presence: true
   validate :dates_check
 
+  accepts_nested_attributes_for :course_subjects, reject_if: ->attributes do
+    attributes[:active_flag] == "0" && attributes[:id].blank?
```
Comment (chuanhtuan) [inline app/models/course.rb]: please google about _destroy
http://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html
http://apidock.com/rails/ActiveRecord/NestedAttributes/ClassMethods/accepts_nested_attributes_for
http://stackoverflow.com/questions/17288774/rails-destroy-method-not-working

File: app/models/course.rb [hunk 6]
```diff
@@ -6,9 +14,32 @@ class Course < ActiveRecord::Base
...
+    course_subjects_hash = Hash.new
+    course_subjects = self.course_subjects
+    course_subjects.each do |course_subject|
+      course_subjects_hash[course_subject.subject_id] = course_subject
+    end
+    subjects = Subject.all
+    return_course_subjects = Array.new
+    if subjects.present?
+      subjects.each do |subject|
+        if course_subjects_hash.has_key? subject.id
```
Comment (chuanhtuan) [inline app/models/course.rb]: subjects.include?(subject)

File: app/views/supervisor/courses/show.html.erb [hunk 7]
```diff
@@ -7,13 +7,39 @@
...
           class: "btn btn-default" %>
         <%= link_to t(:edit), edit_supervisor_course_path(@course),
-          class: "btn btn-success" %>
+          class: "btn btn-primary" %>
         <%= link_to t(:delete), [:supervisor, @course], method: :delete,
           class: "btn btn-danger", data: { confirm: t(:confirm_message) } %>
       </div>
     </div>
     <div class="panel-body">
+    <font size="4">Supervisor's: (0)</font> <%= link_to t(:add_edit_text), '#' %> <br>
```
Comment (chuanhtuan) [inline app/views/supervisor/courses/show.html.erb]: i18n

Comment (MuhammadTamzid) [inline app/views/supervisor/courses/show.html.erb]: Sorry sir. Fixed it in our next pull request

**Pull request merged**